csharp/src/Ice/IBinaryConnection.cs

```diff
@@ -0,0 +1,48 @@
+//
+// Copyright (c) ZeroC, Inc. All rights reserved.
+//
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using ZeroC.Ice.Instrumentation;
+
+namespace ZeroC.Ice
+{
+    public interface IBinaryConnection : IDisposable, IAsyncDisposable
+    {
+        Endpoint Endpoint { get; }
+        ITransceiver Transceiver { get; }
+
+        /// <summary>Gracefully closes the transport.</summary>
+        ValueTask CloseAsync(Exception exception, CancellationToken cancel);
+
+        void IDisposable.Dispose()
+        {
+            GC.SuppressFinalize(this);
```

Comment:
See https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1816?view=vs-2019 and this comment in particular: `To prevent derived types with finalizers from having to reimplement IDisposable and to call it, unsealed types without finalizers should still call GC.SuppressFinalize.`

The other option would be to implement Dispose in each concrete class.